src/ShimSkiaSharp/SKImageFilter.cs

```diff
@@ -52,6 +52,9 @@ public static SKImageFilter CreateOffset(float dx, float dy, SKImageFilter? inpu
     public static SKImageFilter CreatePaint(SKPaint paint, CropRect? cropRect = null) 
         => new PaintImageFilter(paint, cropRect);
 
+    public static SKImageFilter CreateShader(SKShader shader, bool dither, CropRect? cropRect = null)
+        => new ShaderImageFilter(shader, dither, cropRect);
```

Comment:
This shader filter is the new filter that is preferred over paint filter. According to Google: https://github.com/google/skia/commit/7d0f8531580fc07a70e62c33549f64c6db691677

> SkImageFilters::Paint did not use every slot of the SkPaint, with only
> its color, alpha, color filter, and shader having a meaningful effect on
> the image filter result. It was always blended into a transparent dst,
> so blend mode wasn't very relevant, and it was always filled to whatever
> required geometry, so stroke style, path effect, and mask filters were
> ignored or not well specified.
> 
> Color, alpha, and color filter can all be combined into an SkShader, so
> a more constrained SkImageFilters::Shader provides the same useful
> capabilities without as many surprises.

The Paint filter right now may do extra things, but it appears that all the things can be done via merged filters. For example, a fill image/color can be a done with an Image/Color shader. I am not too familiar with how the image filters get created in Svg.Skia and all the potential caveats so I did not feel too comfortable switching just yet.

In 2.x, the Shader filter is really a paint filter with the shader set so should be identical to 3.x Shader. In 3.x, the Paint filter is really a lie and is just the Shader from the paint, so the behavior might be different. https://github.com/mono/SkiaSharp/pull/2789/files#diff-a167a40fc94fe828c0c2d37f6e69d29949af733c40bf8a8b632223a277b2f712R426

I may try be smarter and detect if you just have a Shader, then use a Shader filter, if you just have a Color or something then use that filter type. However, it may be better to just do it right in the caller so SkiaSharp does not do weird things.

If you have ideas on this, I can try and make the compat even more awesome.